active-directory-b2c-dotnet-uwp/MainPage.xaml.cs

```diff
@@ -36,41 +36,36 @@ public MainPage()
         private async void SignInButton_Click(object sender, RoutedEventArgs e)
         {
             AuthenticationResult authResult = null;
+            IEnumerable<IAccount> accounts = await App.PublicClientApp.GetAccountsAsync();
             try
             {
-                authResult = await App.PublicClientApp.AcquireTokenAsync(App.ApiScopes, GetUserByPolicy(App.PublicClientApp.Users, App.PolicySignUpSignIn), UIBehavior.SelectAccount, string.Empty, null, App.Authority);
+                IAccount currentUserAccount = GetUserByPolicy(accounts, App.PolicySignUpSignIn);
+                authResult = await App.PublicClientApp.AcquireTokenSilentAsync(App.ApiScopes, currentUserAccount, App.Authority, false);
```

Comment:
Catching the uirequired exception here.
The call api still shows "unauthorized"

Reply:
Do you get unauthorized for all account types? Are you requesting an AT for the expected scope? (or is the scope/resource) accessed different from the one the AT is acquired for?

---
In reply to: [217924219](https://github.com/Azure-Samples/active-directory-b2c-dotnet-uwp/pull/3#discussion_r217924219) [](ancestors = 217924219)

Reply:
we've been having issues with this. i'll sync with Parakh and see if we can figure it out. might be a settings issue.